src/app/work-item/work-item-list/work-item-detail/work-item-detail.component.ts

```diff
@@ -72,24 +74,53 @@ export class WorkItemDetailComponent implements OnInit {
     this.route.params.forEach((params: Params) => {
       if (params['id'] !== undefined) {
         let id = params['id'];
-        this.workItemService.getWorkItemById(id)
-          .then(workItem => {
-            this.closeRestFields();
-            this.titleText = workItem.attributes['system.title'];
-            this.descText = workItem.attributes['system.description'];
-            this.workItem = workItem;
-            // fetch the list of user 
-            // after getting the Workitem
-            // to set assigned user 
-            // for this workitem from the list
-            this.getAllUsers();
-            this.activeOnList(400);
-          })
-          .catch (err => this.closeDetails());
+        if (id.indexOf('new') >= 0){
+          //Add a new work item          
+          this.addNewWI = true;
+          let type = id.substring((id.indexOf('?') + 1), id.length);
+          this.createWorkItemObj(type);
+          this.getAllUsers();
```

Comment:
This function wouldn't be required following the code on template, but since I think choosing the assignee before saving the item should be implemented, this will be needed.